src/frontend/src/components/ChangesList.tsx

```diff
@@ -4,19 +4,62 @@
  */
 
 import { Link } from '@mui/material';
-import Tooltip from '@mui/material/Tooltip';
+import Tooltip, { tooltipClasses, TooltipProps } from '@mui/material/Tooltip';
 import Typography from '@mui/material/Typography';
 import { ImplementedChange } from 'shared';
 import { fullNamePipe, datePipe } from '../utils/pipes';
 import { Link as RouterLink } from 'react-router-dom';
 import { routes } from '../utils/routes';
 import BulletList from './BulletList';
+import styled from '@emotion/styled';
+import { useState, useLayoutEffect } from 'react';
 
 interface ChangesListProps {
   changes: ImplementedChange[];
 }
 
 const ChangesList: React.FC<ChangesListProps> = ({ changes }) => {
+  const [bodyWidth, setBodyWidth] = useState<number>(window.document.body.offsetWidth);
+
+  window.document.body.addEventListener('resize', () => {
+    setBodyWidth(window.document.body.offsetWidth);
+  });
+
+  function useWindowSize() {
+    const [innerWidth, setInnerWidth] = useState<number>(0);
+    function determinePosition() {
+      return window.innerWidth < bodyWidth ? 'top' : 'right';
+    }
+    const [position, setPosition] = useState<'top' | 'right'>(determinePosition());
+    useLayoutEffect(() => {
+      function updateTooltipProps() {
+        setInnerWidth(window.innerWidth);
+        setPosition(determinePosition());
+      }
+      window.addEventListener('resize', () => {
+        updateTooltipProps();
+      });
+      updateTooltipProps();
+      return () =>
+        window.removeEventListener('resize', () => {
+          updateTooltipProps();
+        });
+    }, []);
+    return [innerWidth, position];
+  }
+
+  let [innerWidth, position] = useWindowSize() as [number, 'top' | 'right'];
```

Comment:
do we need to cast this?

Reply:
When I delete it, when I set the Position for the tooltip, I get this type error. There's probably a better way or something I'm missing but casting was the only thing that resolved it for me
<img width="862" alt="image" src="https://user-images.githubusercontent.com/15839864/217157017-3b5ef1c0-51fa-451d-b384-2f8d982c243d.png">